### DISCUSSION TITLE
save the settings

### DISCUSSION THREAD

**Discussion by DocS788** [ChristopherKlay/StadiaEnhanced#119] [Ideas & Requests]
Hi great Dev,

will be it possible to add a feature saving the settings made in stadiaEnhanced?

Thanks a lot for your work :)

**Comment by ChristopherKlay** [comment]
You mean the Stadia Enhanced specific settings, to carry them between different browsers/profiles and the like?

**Reply by DocS788** [reply]
yes, also in the same profile. For example after some update it does not mantain previous setting, at least for me :)

**Reply by ChristopherKlay** [reply]
This has been the case in the past, if settings change their functionality only.

For example, the "Pro" filter switching to a "Game Labels". If those changes happen, the setting for that specific function is going to default to zero, to avoid hiding content without the users agreement.

Exporting/importing settings is now added to the project board however (:

**Reply by m-cerveny** [reply]
> Exporting/importing settings is now added to the project board however (:
- is this in the works then?
- manual export/import or some automatic (google/user) data space that can be read by another browser instance?

**Reply by ChristopherKlay** [reply]
> is this in the works then?

It is, it's just a lot of work, because i want to support saving settings based on the username/tag and being able to easily import/export your settings (for a specific user) as well.

> manual export/import or some automatic (google/user) data space that can be read by another browser instance?

Likely manual (by just hitting a button) but it will be possible to import/export to any browser running Stadia Enhanced.

**Reply by m-cerveny** [reply]
Wow! That's indeed a lot of work. It's great that you have it in mind but, please, don't be pressed and take your time. You have done a spectacular job!

**Reply by ChristopherKlay** [reply]
A little update.

It's not ready yet and still needs to be tested heavily due to how much can go wrong with these changes, but the current dev build is now slowly getting prepared for `1.16`, which will include;

> * Changed how settings are saved/managed inside the extension
> * Added the ability to import and export all settings of the current profile
> * Changed old/unused settings to be automatically discarded from user profiles

Easily accessible via the menu, instantly applying changes
![image](https://user-images.githubusercontent.com/12472611/127898773-621d9265-f887-4e88-847b-51c4173ac469.png)

You will be able to simply hit "export settings" (translations not final) which downloads a tiny JSON file containing your entire profile and then hit "import settings" in a different browser, or even different profile and it will apply those saved settings.
